### ISSUE TITLE
Missing Network Crate

### ISSUE THREAD

**Issue by zamderax** [madsmtm/objc2#646]
NSNetService in Foundation has all been deprecated:
* See official documentation here: https://developer.apple.com/documentation/foundation/nsnetservice
* See https://forums.developer.apple.com/forums/thread/682744

It's been succeeded by the Network Framework: 
https://developer.apple.com/documentation/network?language=objc

I can only see a `NetworkExtension` autogenerated crate; which is completely different than `Network` 

**Question:** 
1. How can I generate the `Network` crate/feature since it's missing 
2. Alternatively, can I call these methods without the Network crate on version 0.5.2?

**Comment by madsmtm** [comment]
Yeah, see https://github.com/madsmtm/objc2/issues/556, more work is needed to figure out how memory management should work.

> can I call these methods without the Network crate on version 0.5.2?

You _should_ be able to, you'll have to write the `extern "C"` definitions yourself, but it's a lot of work. For an API like [`nw_browser_create`](https://developer.apple.com/documentation/network/nw_browser_create(_:_:)?language=objc), to take a random example, the C definition is:
```objective-c
NW_RETURNS_RETAINED nw_browser_t
nw_browser_create(nw_browse_descriptor_t descriptor,
				  _Nullable nw_parameters_t parameters);
```

So in Rust you'd write:
```rust
// Declare protocols that is used in Network
extern_protocol!(
    unsafe trait OS_nw_browser: NSObjectProtocol {}
    unsafe impl ProtocolType for dyn OS_nw_browser {}
);
type nw_browser_t = ProtocolObject<dyn OS_nw_browser>;

extern_protocol!(
    unsafe trait OS_nw_parameters: NSObjectProtocol {}
    unsafe impl ProtocolType for dyn OS_nw_parameters {}
);
type nw_parameters_t = ProtocolObject<dyn OS_nw_parameters>;

extern_protocol!(
    unsafe trait OS_nw_browse_descriptor: NSObjectProtocol {}
    unsafe impl ProtocolType for dyn OS_nw_browse_descriptor {}
);
type nw_browse_descriptor_t = ProtocolObject<dyn OS_nw_browse_descriptor>;

// The actual function prototype
extern "C" {
    fn nw_browser_create(
        descriptor: &nw_browse_descriptor_t,
        parameters: Option<&nw_parameters_t>,
    ) -> *mut nw_browser_t;
}

// SAFETY: Uses `Retained::from_raw` because of `NW_RETURNS_RETAINED`.
let browser = unsafe { Retained::from_raw(nw_browser_create(descriptor, parameters)) };
```

I _think_, at least, I doubt I'm familiar enough with the Network framework to help you out further, though feel free to ask!

**Comment by roblabla** [comment]
With #556 closed, could a Network framework now be generated?

**Comment by pronebird** [comment]
Yes `nw_retain` and `nw_release` need to be used. 

With addition of Network framework, it should be possible to finally expose the [`virtualInterface`](https://developer.apple.com/documentation/networkextension/nepackettunnelprovider/virtualinterface-9fpgd?language=objc) from `NEPacketTunnelProvider`. This property is available since iOS 18. 

I wonder how availability is being handled by objc2? Is there anything stopping the consumer of objc2 framework from calling functions that aren't available on the deployment target?

**Comment by PaulDance** [comment]
This is becoming important for us because we use [`NetworkExtension`](https://developer.apple.com/documentation/networkextension?language=objc) that deprecated [`NWEndpoint`](https://developer.apple.com/documentation/networkextension/nwendpoint?language=objc) and all its sub-classes since macOS 15 ending at mac0S 15 as well in favor of [`nw_endpoint_t`](https://developer.apple.com/documentation/Network/nw_endpoint_t?language=objc), so in turn deprecates a whole bunch of other types and methods in the framework, while there is currently no easy pure-objc2 way to handle this transition as the `Network` types nor the `nw_endpoint_t`-based methods are made available here.

**Comment by pronebird** [comment]
I have started working on it in https://github.com/madsmtm/objc2/pull/786. Please help me shape it into something workable.
